### ISSUE TITLE
Download button keeps re-downloading the same thing

### ISSUE THREAD

**Issue by MarkJeronimus** [ITDev93/SHM-MOD#6]
I had to download the SHM MOD app three times, not exactly because of a bug, but also not because any fault of mine. It just happens that I tapped somewhere on the screen, at the same time that the "install" popup appeared, which dismissed it.

The first time I thought the phone screen went darker because it would almost turn off the screen, so I tapped (with a sliding motion, because I learned from previous mistakes) to wake it up again. Only the screen wasn't going dark, the app went dark to direct attention to the _tiny_ bar at the bottom.

The second time I downloaded it, (why doesn't it just use the already downloaded apk?) I tapped "Install" (Why does "Dismiss" even exist?) and Android popped up that I need to give the companion app the "Install" permission. When I slid the switch to the right for that app, I pressed back (with the intention to get back to the companion app), only Android happened to show me the "Install unknown application" popup, which was promptly dismissed.

To summarize, I noticed five intertwined download-related issues:

- Downloading can't be canceled and require app kill
- It doesn't show downloaded and total megabytes (unless it precisely happens to be 100MiB)
- Download button doesn't install when it's already downloaded, but re-downloads
- The Install bar is too narrow (just ~6mm on my S21+), about half the height of a normal row, in say, Android Settings
- The Install bar contains a useless "Dismiss" button (why would you want to dismiss if you waited out until it was downloaded and can't re-use the downloaded file)

**Comment by ITDev93** [comment]
Please share a screenshot or a video of what you mean by download issues...

As for the points you mentioned: 
- It never occurred to me that one wants to cancel the download, and if it's a must, then as you said, killing the app will suspend the download, but why even do that...?
- A possible feature implementation to have size if metadata consists of precise size...
- A possible solution is to pop-up message to install or re-download...
- i will consider increasing the snackbar size so buttons are easier to tap...
- sometimes you want to download and share the file, or you want to install it later, thus the dismiss button...

**Comment by MarkJeronimus** [comment]
I don't know how useful a video will be. The steps are pretty clear to me. 

1. Press download
2. dismiss/ignore the snackbar
3. There's no clear way to install the downloaded apk without redownloading.

**Comment by ITDev93** [comment]
> I don't know how useful a video will be. The steps are pretty clear to me. 
> 
> 1. Press download
> 2. dismiss/ignore the snackbar
> 3. There's no clear way to install the downloaded apk without redownloading.

Well, as mentioned, the snackbar has install if you wish to proceed with installation, or dismiss and install later from storage, or perhaps share the apk later...

I could enhance the download, if file exists,  popup install or download again to overwrite the existing apk...

Watck apk snackbar doesn't have install because it is not meant to be installed on the phone...

**Comment by ITDev93** [comment]
Closing this as SHM MOD might reach its end, a full make over will happen to the companion...
